src/components/data/EntitiesContent.js

```diff
@@ -297,7 +298,7 @@ const EntitiesContent = ({
       }, sortedEntities),
     ]);
 
-    if (isSet) {
+    if (isSet && forDownload) {
```

Comment:
This is a bit of a late design change, but having the method change its return type depending on the inputs is a code smell.  I wonder if it'd be easier to test this method and make changes to it if you eliminated the responsibility of generting the zipfile and pushed it back up a layer.

Rather than having `buildTsv` return _either_ an entityTsv _or_ a zipfile, you could have explicit `buildEntityTsv` and `buildMembershipTsv` functions.  The caller could choose to compose them into a ZipFile based on the knowledge that the context is a download.  (That zip combination logic could also be its own method).

The methods then might be more individually testable, and this method ends up going away.

Reply:
Yeah it was pretty obnoxious to stumble upon this situation where it's always returning a zipfile but one is not always needed.  I'm all for this new design but not sure it's worth putting more time into.